core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -485,12 +485,17 @@ object SparkSubmit extends CommandLineUtils {
 
     // In client mode, launch the application main class directly
     // In addition, add the main application jar and any added jars (if any) to the classpath
-    if (deployMode == CLIENT) {
+    // Also add the main application jar and any added jars to classpath in case yarn#client
```

Comment:
s/`yarn#client`/the YARN client/

Otherwise it seems you're talking about yarn-client mode.